src/IJ_mv/IJMatrix_parcsr.c

```diff
@@ -2650,11 +2664,44 @@ hypre_IJMatrixAssembleParCSR(hypre_IJMatrix *matrix)
                   }
                   else if (j != diag_pos[i])
                   {
-                     diag_j[i_diag] = (HYPRE_Int)(local_j[j] - col_0);
+                     diag_j[i_diag] = local_j[j] - col_0;
```

Comment:
Why did you eliminate casting to HYPRE_Int here?

Reply:
This is wrong, probably something that I brought from recmat and forgot to add the casting. I fixed this in 
00aa9cd. Thanks for catching!